### PR TITLE
Publish Docker image to Docker Hub on push to main branch

### PR DESCRIPTION
**Pull request by guilload**
### Description
Publish Docker image to Docker Hub at `quickwit/quickwit:main` on push to main branch. 

### How was this PR tested?
- [x] Built Docker image locally.
- [x] Ensured image is correctly published

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,43 @@
+FROM rust:bullseye AS builder
```
Comment (guilload) [inline Dockerfile:1]: FYI, `bullseye` is the new Debian stable. `rust:bullseye` points to the latest version of Rust on `bullseye`.

File: .github/workflows/publish_main_docker_image.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+name: Publish Docker image to Docker Hub (main)
+
+on:
+  push:
+    branches: [main]
```
Comment (fulmicoton) [inline .github/workflows/publish_main_docker_image.yaml:5]: I'd rather have a manual trigger. What do you think?

File: Dockerfile (hunk 1 above)
Comment (fulmicoton) [reply Dockerfile:1]: thanks for the info

**Review by fulmicoton — APPROVED** [review]

File: .github/workflows/publish_main_docker_image.yaml (hunk 2 above)
Comment (guilload) [reply .github/workflows/publish_main_docker_image.yaml:5]: I like having an automatic build for the `main` branch. That way you know that `quickwit/quickwit:main` always reflects the latest version of `main`. Having a manual trigger for another tag makes sense though. I don't think we'd want our deployments to point to main.

**Pull request merged**